lib/src/code_field/code_controller.dart

```diff
@@ -334,6 +333,56 @@ class CodeController extends TextEditingController {
     );
   }
 
+  /// Modify the rows, that are currently affected by selection.
+  /// Do not take into account `\n` symbols. They are inserted automatically after each row excluding the last one.
+  ///
+  /// Row is considered to be affected by a selection if:
+  /// - The row is completely selected.
+  /// - The start of the selection lies on the row.
+  /// - The end of the selection lies on the row.
+  ///
+  /// [modifierCallback] - transformation function that modifies the row.
+  void modifySelectedRows(String Function(String row) modifierCallback) {
+    final lines = value.text.split('\n');
+
+    int count = 0;
+    int insertedBefore = 0;
+    int insertedWithin = 0;
+    final str = StringBuffer();
+
+    for (int i = 0; i < lines.length; i++) {
```

Comment:
It is too wasteful to go through all lines.
Use `CodeLines.characterIndexToLineIndex()`.